src/amber/cli/generators/auth.cr

```diff
@@ -1,116 +1,19 @@
-require "./generator"
-require "./field"
-
 module Amber::CLI
   class Auth < Generator
-    property fields : Array(Field)
+    command :auth
+    property auth : Generator
 
     def initialize(name, fields)
-      super(name, nil)
-      @fields = all_fields(fields)
-    end
-
-    def pre_render(directory)
-      add_routes
-      add_plugs
-      add_dependencies
-      inject_application_controller_methods
-    end
-
-    private def add_routes
-      add_routes :web, <<-ROUTES
-        get "/profile", #{class_name}Controller, :show
-          get "/profile/edit", #{class_name}Controller, :edit
-          patch "/profile", #{class_name}Controller, :update
-          get "/signin", SessionController, :new
-          post "/session", SessionController, :create
-          get "/signout", SessionController, :delete
-          get "/signup", RegistrationController, :new
-          post "/registration", RegistrationController, :create
-      ROUTES
-    end
-
-    private def add_plugs
-      add_plugs :web, <<-PLUGS
-        plug Authenticate.new
-      PLUGS
-    end
-
-    private def add_dependencies
-      add_dependencies <<-DEPENDENCY
-      require "../src/models/**"
-      require "../src/pipes/**"
-      DEPENDENCY
-    end
-
-    private def inject_application_controller_methods
-      filename = "./src/controllers/application_controller.cr"
-      controller = File.read(filename)
-      append_text = ""
-
-      unless controller.includes? "property current_#{@name}"
-        append_text += current_method_definition
-      end
-
-      unless controller.includes? "def signed_in?"
-        append_text += signed_in_method_definition
-      end
-
-      unless controller.includes? "def redirect_signed_out_#{@name}"
-        append_text += redirect_signed_out_method_definition
+      super(name, fields)
+      if config.model == "crecto"
+        @auth = CrectoAuth.new(name, fields)
```

Comment:
`def initialize(name, fields, orm : Class)` to make the class more flexible and extendable. Right now if someone would like to add auth generator for another orm (Think Jeniffer, LuckyRecord) will need to change this class.

Reply:
This will be going away completely when we move to shards.  Each shard will either override the commands or they will provide new ones.